### PR TITLE
(docs) Add link for instances of Array

### LINKED ISSUE #2088: Refer to both tutorials and API references as links
1. Outside of code blocks, transform all references to standard manual modules, types and functions into links
2. Sentences referring to tutorials matching an standard library module should also contain a direct link to the API reference

Examples:

In [https://ocaml.org/docs/basic-data-types](../blob/main/data/tutorials/language/0it_01_basic_datatypes.md)

Good, links:

```md
Operations on `float` are provided by the [`Stdlib`](/api/Stdlib.html) and the [`Float`](/api/Float.html) modules.
```

Bad, no links:

```md
The predefined function `fst` returns the first element of a pair, while `snd` returns the second element of a pair.
```

### PR DESCRIPTION
**Pull request by rmeis06**
Hello! I'm an outreachy applicant.

I've updated mentions of the Array module to be a link where they were not already. This should partially fix https://github.com/ocaml/ocaml.org/issues/2088

@cuihtlauac  @SaySayo Does this look correct to yinz?

### CODE REVIEW

**Review by christinerose — APPROVED** [review]

File: data/tutorials/language/0it_01_basic_datatypes.md
Comment (cuihtlauac) [inline data/tutorials/language/0it_01_basic_datatypes.md:1]: These occurences of the string `Array` do not refer to the Array module from the standard library but to the Array constructor (line 566) from the type json (line 560). Links should not be added here

File: data/tutorials/language/0it_05_imperative.md
Comment (cuihtlauac) [inline data/tutorials/language/0it_05_imperative.md:1]: This one is tricky, the two occurences of Array here refer to the module defined line 743, not the module from the standard library, they should not be changed.

File: data/tutorials/language/1ms_01_functors.md [hunk 1]
```diff
@@ -477,7 +477,7 @@ Run the `dune utop` command. Once inside the toplevel, enter the following comma
 - : int list = [0; 1; 3; 6; 10; 15; 21; 28; 36; 45]
 ```
 
-Modules `Array` and `List` appear augmented with `Array.scan_left` and `List.scan_left`. For brevity, the output of the first two toplevel commands is not shown here.
+Modules [`Array`](/api/Array.html) and `List` appear augmented with `Array.scan_left` and `List.scan_left`. For brevity, the output of the first two toplevel commands is not shown here.
```
Comment (cuihtlauac) [inline data/tutorials/language/1ms_01_functors.md]:
```suggestion
Modules `Array` and `List` appear augmented with `Array.scan_left` and `List.scan_left`. For brevity, the output of the first two toplevel commands is not shown here.
```

This one does not refer to the version from the standard library. No link should be added.

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Comment by rmeis06** [comment]
@cuihtlauac I believe I have the documents updated as described. A question for me to better understand ocaml, a standard library can be overwritten in a local program using `module <module name>`? And I think I'll have to learn more about what a constructor is to understand the Array that was defined in json.

**Review by cuihtlauac — APPROVED** [review]
Thanks @rmeis06

**Pull request merged**